crates/wash/src/cli/mod.rs

```diff
@@ -259,17 +258,91 @@ impl CommandOutput {
     }
 }
 
+/// Non-generic copy of etcetera::AppStrategy trait to make it dyn-compatible
+pub trait DirectoryStrategy: std::fmt::Debug + Send + Sync {
+    fn home_dir(&self) -> &Path;
+    fn config_dir(&self) -> PathBuf;
+    fn data_dir(&self) -> PathBuf;
+    fn cache_dir(&self) -> PathBuf;
+    fn state_dir(&self) -> Option<PathBuf>;
+    fn runtime_dir(&self) -> Option<PathBuf>;
+    fn in_config_dir(&self, path: &str) -> PathBuf {
+        let mut p = self.config_dir();
+        p.push(Path::new(&path));
+        p
+    }
+    fn in_data_dir(&self, path: &str) -> PathBuf {
+        let mut p = self.data_dir();
+        p.push(Path::new(&path));
+        p
+    }
+    fn in_cache_dir(&self, path: &str) -> PathBuf {
+        let mut p = self.cache_dir();
+        p.push(Path::new(&path));
+        p
+    }
+    fn in_state_dir(&self, path: &str) -> Option<PathBuf> {
+        let mut p = self.state_dir()?;
+        p.push(Path::new(&path));
+        Some(p)
+    }
+    fn in_runtime_dir(&self, path: &str) -> Option<PathBuf> {
+        let mut p = self.runtime_dir()?;
+        p.push(Path::new(&path));
+        Some(p)
+    }
+}
+
+impl DirectoryStrategy for Xdg {
+    fn home_dir(&self) -> &Path {
+        <Xdg as AppStrategy>::home_dir(self)
+    }
+    fn config_dir(&self) -> PathBuf {
+        <Xdg as AppStrategy>::config_dir(self)
+    }
+    fn data_dir(&self) -> PathBuf {
+        <Xdg as AppStrategy>::data_dir(self)
+    }
+    fn cache_dir(&self) -> PathBuf {
+        <Xdg as AppStrategy>::cache_dir(self)
+    }
+    fn state_dir(&self) -> Option<PathBuf> {
+        <Xdg as AppStrategy>::state_dir(self)
+    }
+    fn runtime_dir(&self) -> Option<PathBuf> {
+        <Xdg as AppStrategy>::runtime_dir(self)
+    }
+}
+
+impl DirectoryStrategy for Windows {
+    fn home_dir(&self) -> &Path {
+        <Windows as AppStrategy>::home_dir(self)
+    }
+    fn config_dir(&self) -> PathBuf {
+        <Windows as AppStrategy>::config_dir(self)
+    }
+    fn data_dir(&self) -> PathBuf {
+        <Windows as AppStrategy>::data_dir(self)
+    }
+    fn cache_dir(&self) -> PathBuf {
+        <Windows as AppStrategy>::cache_dir(self)
+    }
+    fn state_dir(&self) -> Option<PathBuf> {
+        <Windows as AppStrategy>::state_dir(self)
+    }
+    fn runtime_dir(&self) -> Option<PathBuf> {
+        <Windows as AppStrategy>::runtime_dir(self)
+    }
+}
+
 /// CliContext holds the global context for the wash CLI, including output kind and directories
 ///
 /// It is used to manage configuration, data, and cache directories based on the XDG Base Directory Specification,
 /// or a custom configuration if needed.
 #[derive(Debug, Clone)]
 pub struct CliContext {
-    // TODO(#25): Just store an Arc-ed trait object
-    #[cfg(unix)]
-    app_strategy: Xdg,
-    #[cfg(windows)]
-    app_strategy: Windows,
+    /// Application strategy to access configuration directories.
+    app_strategy: Arc<dyn DirectoryStrategy>,
```

Comment:
This indeed requires an `Arc` as `Xdg` and `Windows` do not implement `Clone`. So using a `Box` would require some wrapper object/enum with smart `Clone`ing implemented.